repair/repair.cc

```diff
@@ -1656,7 +1657,9 @@ future<> bootstrap_with_repair(seastar::sharded<database>& db, seastar::sharded<
                         };
                         auto old_endpoints_in_local_dc = get_old_endpoints_in_local_dc();
                         auto rf_in_local_dc = get_rf_in_local_dc();
-                        if (old_endpoints.size() == strat.get_replication_factor()) {
+                        if (everywhere_topology) {
+                            neighbors = old_endpoints_in_local_dc;
```

Comment:
Does this stream the data from *all* the nodes - potentially a hundred of them? Is this what we want to do?

Reply:
We do not need to stream all the same data N  times. We need to sync between all the nodes with repair though. In practice, I doubt real user would use everywhere topology to store massive data. Another point is that user uses few fat nodes instead of hundreds of tiny nodes.

Reply:
Then why repair from endpoints only in local_dc rather than everywhere, i.e. in other dc's as well?

I'm not sure what are the defined/desired semantics of EverywhereStrategy.
https://docs.datastax.com/en/dse/5.1/cql/cql/cql_using/cqlKeyspacesAbout.html
```
EveryWhereStrategy is a specialized DataStax Enterprise strategy used by the dse_system keyspace and is not intended for customer use.
```

And https://support.datastax.com/hc/en-us/articles/208026816-DSE-EverywhereStrategy-is-not-understood-by-Apache-Cassandra-non-DSE-nodes-and-can-cause-restart-failures proposes:
```
Workaround
The best approach is to keep the DSE node in the cluster and then alter the replication strategy for the dse_system keyspace to one that is understood by all nodes, for example:

ALTER KEYSPACE dse_system WITH replication = {'class': 'NetworkTopologyStrategy', 'DC1':1 ,'DC2':1};

```

So the recommendation above seems to indicate that it refers to nodes on all dc's.

Reply:
How do we define it? Is it documented anywhere? (it should be wither in the source tree if it's internal or in the user docs if it's exposed to users)

Reply:
The most safe way is to stream from all nodes from all dcs. It is comprise between speed and safety to stream from local dcs only. This is also consistent with network_topology where bootstrap streams from local dc.

I do not think we even want to allow user to use the  EverywhereStrategy at all. How we choose nodes to stream from is completely internal and should not be exposed to users. 

I will update docs/design-notes/repair_based_node_ops.md.

Reply:
PR sent: https://github.com/scylladb/scylla/pull/8518